client/asset/btc/rpcclient.go

```diff
@@ -157,6 +171,17 @@ func (wc *rpcClient) connect(ctx context.Context, _ *sync.WaitGroup) error {
 	if codeVer < minProtocolVersion {
 		return fmt.Errorf("node software out of date. version %d is less than minimum %d", codeVer, minProtocolVersion)
 	}
+	chainInfo, err := wc.getBlockchainInfo()
+	if err != nil {
+		return fmt.Errorf("getblockchaininfo error: %w", err)
+	}
+	if !chainOK(wc.cloneParams.Network, chainInfo.Chain) {
+		return errors.New("wrong net")
+	}
+	// Ensure we can get as well as decode an address from the wallet.
+	if _, err = wc.externalAddress(); err != nil {
+		return fmt.Errorf("address error: %w", err)
+	}
```

Comment:
I'm not sure at all, but does this create a danger of going over the gap limit for these wallets from multiple connections with no use?

Reply:
Fair point.  It would take doing that 1000x I think, but still.

Actually, I'm not seeing why `(*xcWallet).Connect` would not error when getting or validating deposit address.

Reply:
Ah, `btcutil.DecodeAddress` does not check that segwit addresses' hrp of the prefix is for the intended network.  We have to do that: https://github.com/decred/dcrdex/pull/2326/commits/2302618501d840de6b7066c8c51c845915c26d2f
Removed the `externalAddress` request from `connect` since it is redundant with `(*xcWallet).Connect`'s call to get a deposit address if there isn't one.  Now it's only `reconfigure` that tries to get and check a new address, and even that is not fatal, instead just requiring a full wallet restart.